pkg/kube/inject/webhook.go

```diff
@@ -432,6 +432,32 @@ func escapeJSONPointerValue(in string) string {
 	return strings.Replace(step, "/", "~1", -1)
 }
 
+func updateLabels(target map[string]string, added map[string]string) (patch []rfc6902PatchOperation) {
+	for key, value := range added {
+		if target == nil {
```

Comment:
optional nitpick, I would do

```golang
patches := [] rfc6902PatchOperation{}
for {
  patch := rfc6902PatchOperation{...}
  if target == nil {
  } else if target[key] == "" {
  } else if target[key] == "false" {
  } else  {
  } 
  patches = append(patches, patch)
}
```

each branch set only different parts of the patch field.

Reply:
@incfly I agree that looks cleaner - but does that keep the method reusable for other labels?  Currently `mtlsReady` is the only label Istio injects but do we want to keep the method such that it works for other labels down the line?

Reply:
IMO we don't need to sacrifice readability for premature future changes. People making that changes can adapt the code correspondingly by then.